### ISSUE TITLE
Differentiable version of FNMR@FMR metric to use it as loss

### ISSUE THREAD

**Issue by AlekseySh** [OML-Team/open-metric-learning#246]

**Comment by AlekseySh** [comment]
@deepslug Nope, not enough resources. Do you want to try it? If so, the idea is that we want to adapt the approach from this [paper](https://paperswithcode.com/paper/recall-k-surrogate-loss-with-large-batches) and make FNMR@FMR metric differentiable.

**Comment by deepslug** [comment]
Unfortunately, I don't have enough resources either, but this is something I’d bring on top of my (and hopefully your) list!

**Comment by AlekseySh** [comment]
Got it!

**Comment by deepslug** [comment]
As I mentioned in my previous comment on the [post](https://github.com/OML-Team/open-metric-learning/pull/549), I work in the field of biometrics and am keen on seeing the differential version of FNMR@FMR as it can directly optimize the metric. Given the recent active development of OML, I wanted to add this comment to bump up the thread :)

**Comment by AlekseySh** [comment]
@deepslug thank you for you comment. I'd like to add that we've already implemented similar idea in OML. There is [SurrogatePricisonLoss](https://open-metric-learning.readthedocs.io/en/latest/contents/losses.html#surrogateprecision) -- differentiable version of Precision metric. Experiments showed it was able to perform on SOTA level. So, it would be interesting to apply similar idea to FNMR@FMR. 

Contributors are welcome!
